### PR TITLE
Fix nav menus option schema

### PR DESCRIPTION
**Pull request by Chouby**
Fixes https://github.com/polylang/polylang-pro/issues/2212
The schema was missing the locations keys in the complex object `[themes][locations][languages] => menu_id`.
I removed the context as it is already added globally.
I removed the `phpstan-return` as it doesn't help.
Tests are modified accordingly.

### CODE REVIEW

File: include/Options/Business/Nav_Menus.php [hunk 1]
```diff
@@ -45,36 +45,23 @@ protected function get_default() {
...
 				'^\\w+$' => array( // Any word characters as key, correspond to a theme slug.
-					'type'              => 'object',
-					'context'           => array( 'edit' ),
-					'patternProperties' => array(
-						'^[a-z_-]+$' => array( // Language slug as key.
-							'type'    => 'integer',
-							'minimum' => 0, // A post ID.
+					'type'                 => 'object',
+					'patternProperties'    => array(
+						'^\\w+$' => array( // Any word characters as key, correspond to a nav menu location.
```
Comment (Hug0-Drelon) [inline include/Options/Business/Nav_Menus.php:56]: Looks like it accept special char as well, isn't it?
See https://github.com/WordPress/wordpress-develop/blob/0ac1eb312aeb5898c3037e68d1186658236f76d8/src/wp-includes/nav-menu.php#L95

**Review by Hug0-Drelon — CHANGES_REQUESTED** [review]

File: include/Options/Business/Nav_Menus.php (hunk 1 above)
Comment (Chouby) [reply include/Options/Business/Nav_Menus.php:56]: I stupidly copied what was done for the theme slug, but looking at this with more attention now, I think both are wrong and we should use `[\w-]+` here for the location - see https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/rest-api/endpoints/class-wp-rest-menu-locations-controller.php#L53 - and `[^\/:<>\*\?"\|]+` for the theme slug - see https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L23.

Comment (Hug0-Drelon) [reply include/Options/Business/Nav_Menus.php:56]: Indeed, better with what you've found! 👍

File: include/Options/Business/Nav_Menus.php [hunk 2]
```diff
@@ -45,36 +45,23 @@ protected function get_default() {
...
 			'type'                 => 'object', // Correspond to associative array in PHP, @see{https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#primitive-types}.
 			'patternProperties'    => array(
-				'^\\w+$' => array( // Any word characters as key, correspond to a theme slug.
-					'type'              => 'object',
-					'context'           => array( 'edit' ),
-					'patternProperties' => array(
-						'^[a-z_-]+$' => array( // Language slug as key.
-							'type'    => 'integer',
-							'minimum' => 0, // A post ID.
+				'[^\/:<>\*\?"\|]+' => array( // Excludes invalid directory name characters @see https://developer.wordpress.org/reference/classes/wp_rest_themes_controller/register_routes/
```
[the remark this answers is not in the document]
Comment (Chouby) [reply include/Options/Business/Nav_Menus.php:53]: Do you mean copy the whole pattern (and support subdirectories: I wasn't aware of this possibility and didn't find doc about that) or use the constant itself? I found out that WP itself doesn't reuse it :(. See [WP_REST_Themes_Controller::PATTERN](https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php#L69) or https://github.com/WordPress/WordPress/blob/6.6.1/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L108

[the remark this answers is not in the document]
Comment (Hug0-Drelon) [reply include/Options/Business/Nav_Menus.php:53]: Oh no WordPress 😢 
Well, the code you mentioned for Site Editor type of things, so I guess a whole other team has worked on that and wasn't aware of the constant.

That said, if it work without the subdirectories as of today, I guess we can keep it that way. The simpler the better 😅

**Review by Hug0-Drelon — APPROVED** [review]

File: include/Options/Business/Nav_Menus.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chouby) [reply include/Options/Business/Nav_Menus.php:53]: I just found a starting point https://core.trac.wordpress.org/ticket/54349.

**Pull request merged**